src/main/java/export/Exporter.java

```diff
@@ -20,34 +19,37 @@
  */
 public class Exporter {
 
-  public static void export(Table db, Writer writer, Settings settings) throws IOException {
-    List<Column> temp = settings.getColumns(); // Get all the columns
-    List<String> columns = new ArrayList<String>();
-    for (Column col : temp) {
-      columns.add(col.getName());
+  public static void export(Table db, Writer writer) throws IOException {
+
+    Set<String> keys = new TreeSet<String>();
+    for (Record r : db) {
+      keys.addAll(r.keySet());
     }
 
-    CSVWriter csvwriter = new CSVWriter(writer, settings.getDelimiter().charAt(0));
-    csvwriter.writeNext(columns.toArray(new String[columns.size()])); // write
+    CSVWriter csvwriter = new CSVWriter(writer, ';');
+    csvwriter.writeNext(keys.toArray(new String[keys.size()])); // write
     // column
     // names
+
     for (Record dr : db) {
-      csvwriter.writeNext(generateRow(dr, columns));
+      csvwriter.writeNext(generateRow(dr, keys));
     }
     csvwriter.close();
   }
 
-  public static String[] generateRow(Record dr, List<String> columns) {
+  public static String[] generateRow(Record dr, Set<String> columns) {
     List<String> items = new ArrayList<String>();
     for (String column : columns) {
       Value value = dr.get(column);
       if (value == null) {
```

Comment:
Is never used, `value` may be equal to `NullValue` which `value` field is `null`, but `value` itself is never `null`